applications/crossbar/src/modules/provisioner_v5.erl

```diff
@@ -113,21 +98,74 @@ update_account(AccountId, JObj, AuthToken) ->
              ,'none'
             ).
 
-%%--------------------------------------------------------------------
-%% @private
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
--spec provision_data(wh_json:object()) -> wh_json:object().
-provision_data(JObj) ->
-    Routines =
-        [fun set_realm/1
-         ,fun set_owner/1
-         ,fun maybe_set_timezone/1
-         ,fun create_provision_settings/1
-        ],
-    lists:foldl(fun(F, J) -> F(J) end, JObj, Routines).
+%% @public
+-spec update_user(ne_binary(), wh_json:object(), ne_binary()) -> 'ok'.
+update_user(AccountId, JObj, AuthToken) ->
+    case wh_json:get_value(<<"pvt_type">>, JObj) of
+        <<"user">> ->
+            save_user(AccountId, JObj, AuthToken);
+        <<"device">> ->
+            save_user(AccountId, JObj, AuthToken);
+        _ -> ok %% Gets rid of VMbox
+    end.
+
+-spec save_user(ne_binary(), wh_json:object(), ne_binary()) -> 'ok'.
+save_user(AccountId, JObj, AuthToken) ->
+    Realm   = wh_json:get_value(<<"realm">>, set_realm(JObj)),
```

Comment:
Please do not add the realm to the provisioner device, this should only be present on requests to:

POST accounts/{account_id}